storage/filestore.go

```diff
@@ -221,7 +221,7 @@ func (f *FilesystemStore) Set(name string, meta []byte) error {
 	os.RemoveAll(fp)
 
 	// Write the file to disk
-	if err = ioutil.WriteFile(fp, meta, f.perms); err != nil {
+	if err = ioutil.WriteFile(fp, meta, f.perms&notary.PermExecutableMask); err != nil {
```

Comment:
@HuKeping what do you think of @endophage's constants proposal?

With these constants, we could separate `.perms` to `.filePerms` and `.dirPerms` that are initialized to the appropriate constants (`PubExecPerms+PubNoExecPerms` or `PrivExecPerms+PrivNoExecPerms`) depending on which store it is?

I'm worried that adding bitwise masks makes it too easy to make a mistake or forget another place where we need to use it, especially when we're only writing files/directories with one of four permissions.